src/section.ts

```diff
@@ -152,8 +152,8 @@ export function renderSection(props: {
             ${shouldShowLabel
               ? html`<div class="label" style=${styleMap(labelStyle)}>
                   ${show_states && isNotPassthrough
-                    ? html`<span class="state">${formattedState}</span>${show_units
-                          ? html`<span class="unit">${box.unit_of_measurement}</span>`
+                    ? html`<span class="state">${formattedState}</span>${(show_units && convert_units_to != 'monetary')
+                          ? html`&nbsp;<span class="unit">${box.unit_of_measurement}</span>`

```

Comment:
This space and the one above shouldn't be part of this PR. The convention is that unit of measurement is added without space and users can add space with card-mod and CSS if they wish.

Reply:
Fixed